apps/algorithm/recommend/src/recommender.py

```diff
@@ -0,0 +1,141 @@
+import pandas as pd
+import numpy as np
+import tensorflow as tf
+import requests
+import os
+
+
+class Recommender:
+    def __init__(self):
+        self.load_model()
+
+    def download_file(self, url, destination):
+        response = requests.get(url)
+        response.raise_for_status()
+        with open(destination, "wb") as f:
+            f.write(response.content)
+
+    def load_model(self):
+        try:
+            self.data = pd.read_csv("/app/src/training/processed_data.csv")
+            self.cosine_similarity_matrix = np.load(
+                "/app/src/training/cosine_similarity_matrix.npy"
+            )
+            self.normalized_item_vectors = np.load(
+                "/app/src/training/normalized_item_vectors.npy"
+            )
+        except FileNotFoundError:
+            os.makedirs("/app/src/training", exist_ok=True)
+            data_url = "https://github.com/UVicMartletplace/martletplace/releases/download/recommender-v1.0.1/processed_data.csv"
+            cosine_url = "https://github.com/UVicMartletplace/martletplace/releases/download/recommender-v1.0.1/cosine_similarity_matrix.npy"
+            item_vectors_url = "https://github.com/UVicMartletplace/martletplace/releases/download/recommender-v1.0.1/normalized_item_vectors.npy"
+
+            self.download_file(data_url, "/app/src/training/processed_data.csv")
+            self.download_file(
+                cosine_url, "/app/src/training/cosine_similarity_matrix.npy"
+            )
+            self.download_file(
+                item_vectors_url, "/app/src/training/normalized_item_vectors.npy"
+            )
+
+            self.data = pd.read_csv("/app/src/training/processed_data.csv")
+            self.cosine_similarity_matrix = np.load(
+                "/app/src/training/cosine_similarity_matrix.npy"
+            )
+            self.normalized_item_vectors = np.load(
+                "/app/src/training/normalized_item_vectors.npy"
+            )
```

Comment:
Why don't we just recursively call it?
```suggestion
            self.load_model()
```